stats/stats.go

```diff
@@ -308,6 +311,50 @@ func (e *HistogramData) CalcPercentile(percentile float64) float64 {
 	return e.Max // not reached
 }
 
+func sumOfSquares(stdev float64, sum float64, count int64) float64 {
+	fC := float64(count)
+	variance := math.Pow(stdev, 2)
+
+	ret := (variance * fC) + (sum * sum / fC)
+	return ret
+}
+
+func indexSlice(slice []int32, value int32) int {
+	for p, v := range slice {
+		if v == value {
+			return p
+		}
+	}
+	return -1
+}
+
+// Import translate the external representation of the histogram data in
+// an internally usable one.
+func (e *HistogramData) Import() *Histogram {
+	res := NewHistogram(e.Offset,
+		e.Divider)
+	res.Counter.Count = e.Count
+	res.Counter.Min = e.Min
+	res.Counter.Max = e.Max
+	res.Counter.Sum = e.Sum
+	res.Counter.sumOfSquares = sumOfSquares(e.StdDev, e.Sum, e.Count)
+
+	for idx, bucket := range e.Data {
+		if idx < len(e.Data)-1 {
```

Comment:
I don't think you can use the idx ? the exported data can have holes (it doesn't export 0s) while the internal data structure doesn't

Reply:
But this idx is looping over the external data structure and using the offset and divider filling in the histogram at the respective places. So even if there is a hole it shouldn't affect it.